### ISSUE TITLE
shadow warnings from fmt in gcc 10

### ISSUE THREAD

**Issue by madeso** [gabime/spdlog#1556]
With this code:
```cpp
#include "spdlog/spdlog.h"

int
main(int, char* [])
{
    return 0;
}
```
Running `g++ -I. -std=c++17 -Wshadow main.cc` with g++ 10.1.0 on arch linux using spdlog [1.6.0](https://github.com/gabime/spdlog/releases/tag/v1.6.0) I get the following shadow warnings:

```
In file included from ./spdlog/fmt/fmt.h:20,
                 from ./spdlog/common.h:36,
                 from spdlog/spdlog.h:12,
                 from main.cc:1:
./spdlog/fmt/bundled/core.h: In member function ‘const T& fmt::v6::internal::dynamic_arg_list::push(const Arg&)’:
./spdlog/fmt/bundled/core.h:1251:10: warning: declaration of ‘node’ shadows a member of ‘fmt::v6::internal::dynamic_arg_list’ [-Wshadow]
 1251 |     auto node = std::unique_ptr<typed_node<T>>(new typed_node<T>(arg));
      |          ^~~~
./spdlog/fmt/bundled/core.h:1231:37: note: shadowed declaration is here
 1231 |   template <typename = void> struct node {
      |                                     ^~~~
In file included from ./spdlog/fmt/fmt.h:21,
                 from ./spdlog/common.h:36,
                 from spdlog/spdlog.h:12,
                 from main.cc:1:
./spdlog/fmt/bundled/format.h: In member function ‘decltype (ctx.out()) fmt::v6::formatter<fmt::v6::bytes>::format(fmt::v6::bytes, FormatContext&)’:
./spdlog/fmt/bundled/format.h:3247:40: warning: declaration of ‘writer’ shadows a global declaration [-Wshadow]
 3247 |     internal::basic_writer<range_type> writer(range_type(ctx.out()));
      |                                        ^~~~~~
./spdlog/fmt/bundled/format.h:2737:7: note: shadowed declaration is here
 2737 | using writer FMT_DEPRECATED_ALIAS = internal::writer;
      |       ^~~~~~
```

I am 99.99% sure this is a fmt issue, and not a spdlog issue, but since I don't use fmt directly I figured I'd create a 'tracking issue' for upgrading and/or fixing the fmt issue

**Comment by gabime** [comment]
Right, its a fmt issue. Maybe it worth checking if his was already fixed in 6.2.1 (spdlog use 6.2.0)

**Comment by gcerretani** [comment]
It has ben fixed in master, but unfortunately not yet released.
